semantics/language/common/typing/compatibility.k

```diff
@@ -352,8 +352,10 @@ module C-TYPING-COMPATIBILITY
           => t(Mods, functionType(T, P'))
      rule compositeFunctionType(T:Type, Mods:Set, .K, .List, P:List, _)
```

Comment:
Isn't it nicer/faster to write such rules using local rewrites, like:

    rule compositeFunctionType(T:Type, Mods:Set, .K, .List, `P:List => .List`, `_ => P`)
    requires P =/=K .List